reframe/frontend/executors/policies.py

```diff
@@ -416,7 +416,8 @@ def exit(self):
         pollrate = PollRateFunction(0.2, 60)
         num_polls = 0
         t_start = datetime.now()
-        while (self._running_tasks or self._waiting_tasks):
+        while (self._running_tasks or self._waiting_tasks or
+               self._completed_tasks or sum(self._ready_tasks.values(), [])):
```

Comment:
Check my comment about the way to compute the sum of all tasks [here](https://github.com/eth-cscs/reframe/pull/1402#discussion_r456455685). And do it through a separate function.